### PR TITLE
0x36 CALLDATASIZE opcode

### PR DESCRIPTION
**Pull request by richwarner**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: 100

## What is the new behavior?

- Implemented opcode 0x36 CALLDATASIZE

### CODE REVIEW

File: tests/test_basic.py [hunk 1]
```diff
@@ -287,6 +287,14 @@ async def test_environmental_information(self):
         self.assertEqual(res.result.top_stack, Uint256(0, 0))
         self.assertEqual(res.result.top_memory, Uint256(0, 0))
 
+        # CALLDATASIZE
+        code, calldata = get_case(case="./tests/cases/024.json")
+        res = await self.zk_evm.execute(code=code, calldata=calldata).execute(
+            caller_address=1
+        )
+        self.assertEqual(res.result.top_stack, Uint256(0, 0))
```
Comment (abdelhamidbakhta) [inline tests/test_basic.py:295]: Can you please add other tests with non empty calldata ?

**Review by abdelhamidbakhta — APPROVED** [review]
Left comment
